src/libraries/Microsoft.PowerFx.Interpreter/ReflectionFunction.cs

```diff
@@ -207,21 +209,36 @@ private FunctionDescr Scan()
                 info.RetType = GetType(m.ReturnType);
 
                 var paramTypes = new List<FormulaType>();
-                foreach (var p in m.GetParameters())
+
+                info._isAsync = m.ReturnType.BaseType == typeof(Task);
+
+                var parmas = m.GetParameters();
+                for (var i = 0; i < parmas.Length; i++)
                 {
-                    if (typeof(FormulaValue).IsAssignableFrom(p.ParameterType))
+                    if (i == parmas.Length - 1 && info._isAsync)
                     {
-                        paramTypes.Add(GetType(p.ParameterType));
-                    } 
-                    else if (p.ParameterType == ConfigType)
+                        if (parmas[i].ParameterType != typeof(CancellationToken))
+                        {
+                            throw new InvalidOperationException($"Last argument must be a cancellation token.");
```

Comment:
>"Last argument must be a cancellation token."); [](http://example.com/codeflow?start=65&length=47)

Can we add a test that verifies we hit this?  
Ie, does _not_ have a cancellation token:

Task<StringFormula> Execute();

Reply:
Added the test